### PR TITLE
PSP-6952 disposition list view

### PR DESCRIPTION
**Pull request by asanchezr**
disposition-list

### CODE REVIEW

**Comment by devinleighsmith** [comment]
The following are issues with the ui:

image
1) Should be "Team member" not "Type to search..."
2) Should be "Select file status" not "All Status"
3) Should be "Select disposition status" not "All Status"
4) Should be Select disposition type" not "All Type"
5) The words "historical file" should be replaced with "reference"

**Comment by devinleighsmith** [comment]
The team member search doesn't appear to be a multi-select? at least, I wasn't able to select multiple values @asanchezr

**Comment by asanchezr** [comment]
> The team member search doesn't appear to be a multi-select? at least, I wasn't able to select multiple values @asanchezr

@devinleighsmith please see the JIRA story. Requirements were updated to a single select (like we do in ACQ File List). The UI mockup is out of date

**Comment by asanchezr** [comment]
> The following are issues with the ui:
> 
> image
> 
> 1. Should be "Team member" not "Type to search..."
> 2. Should be "Select file status" not "All Status"
> 3. Should be "Select disposition status" not "All Status"
> 4. Should be Select disposition type" not "All Type"
> 5. The words "historical file" should be replaced with "reference"

You are right about the team member typeahead - will change the placeholder.
For the others, while I agree that the placeholder in the UX pin is more readable. I went with what was specifically called out in the story and Confluence. I will post a comment on the story to ask which way BAs prefer

image

**Comment by devinleighsmith** [comment]
changing the numbering to 5 (instead of 10) doesn't seem to work properly.

File: source/backend/dal/Repositories/DispositionFileRepository.cs [hunk 1]
```diff
@@ -162,6 +188,109 @@ public long GetRowVersion(long id)
...
+                .Include(tm => tm.PimsDispositionFileTeams)
+                    .ThenInclude(c => c.DspFlTeamProfileTypeCodeNavigation)
+                .Include(fp => fp.PimsPropertyDispositionFiles)
+                    .ThenInclude(prop => prop.Property)
+                    .ThenInclude(ad => ad.Address)
+                    .ThenInclude(x => x.ProvinceState)
+                .Include(fp => fp.PimsPropertyDispositionFiles)
+                    .ThenInclude(prop => prop.Property)
+                    .ThenInclude(ad => ad.Address)
+                    .ThenInclude(x => x.Country)
```
Comment (devinleighsmith) [inline source/backend/dal/Repositories/DispositionFileRepository.cs:286]: do we need the country?

File: source/frontend/src/features/disposition/list/DispositionSearchResults/DispositionSearchResults.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  const { results, sort = {}, setSort, setPageSize, setPageIndex, totalItems, ...rest } = props;
+
+  // This will get called when the table needs new data
+  const updateCurrentPage = useCallback(
+    ({ pageIndex }: { pageIndex: number }) => setPageIndex && setPageIndex(pageIndex),
+    [setPageIndex],
+  );
+
+  return (
+    <Table<DispositionSearchResultModel>
```
Comment (devinleighsmith) [inline source/frontend/src/features/disposition/list/DispositionSearchResults/DispositionSearchResults.tsx:36]: the cause of the pagination issue may be that you are missing the manual pagination flag.

File: source/frontend/src/features/disposition/list/styles.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  background-color: ${({ theme }) => theme.css.filterBoxColor};
+`;
+
+export const LongInlineInput = styled(InlineInput)`
+  flex: 3 1 auto;
+  max-width: 31rem;
+`;
+
+export const ColButtons = styled(Col)`
+  border-left: 2px solid white;
```
Comment (devinleighsmith) [inline source/frontend/src/features/disposition/list/styles.ts:46]: px -> rem

**Review by devinleighsmith — APPROVED** [review]

File: source/backend/dal/Repositories/DispositionFileRepository.cs [hunk 4]
```diff
@@ -162,6 +188,109 @@ public long GetRowVersion(long id)
...
+
+        public List<PimsDispositionFileTeam> GetTeamMembers()
+        {
+            var predicate = PredicateBuilder.New<PimsDispositionFileTeam>(disp => true);
+
+            return Context.PimsDispositionFileTeams.AsNoTracking()
+                .Include(x => x.DispositionFile)
+                .Include(x => x.Person)
+                .Include(x => x.Organization)
+                .Where(predicate)
```
[the remark this answers is not in the document]
Comment (asanchezr) [reply source/backend/dal/Repositories/DispositionFileRepository.cs:200]: copy-pasta from acq repo - will remove.

File: source/backend/dal/Repositories/DispositionFileRepository.cs (hunk 1 above)
Comment (asanchezr) [reply source/backend/dal/Repositories/DispositionFileRepository.cs:286]: I don't think so - will remove

**Pull request merged**